runtime/mbstring.cpp

```diff
@@ -7,7 +7,31 @@
 #include "common/unicode/unicode-utils.h"
 #include "common/unicode/utf8-utils.h"
 
+bool is_detect_incorrect_encoding_names_warning = false;
+
+void f$set_detect_incorrect_encoding_names_warning(bool show) {
+  is_detect_incorrect_encoding_names_warning = show;
+}
+
+static void detect_incorrect_encoding_names(const string &encoding) {
```

Comment:
This function should be `static int mb_detect_encoding_new()` and return a number. It should not fire any error